include/maths/analytics/CBoostedTreeImpl.h

```diff
@@ -237,14 +237,15 @@ class MATHS_ANALYTICS_EXPORT CBoostedTreeImpl final {
     //! Tag progress through initialization.
     enum EInitializationStage {
         E_NotInitialized = 0,
-        E_SoftTreeDepthLimitInitialized = 1,
```

Comment:
This will invalidate persisted state. One option would be to bump the state when we restore. Alternatively, we could increment the initialisation stage. Since this is still on a feature branch, I'm not sure this should be done now, but maybe we should capture this in an issue so it isn't forgotten.